space2stats_api/src/space2stats_ingest/main.py

```diff
@@ -29,17 +31,14 @@ def load_parquet_to_db(
         connection_string (str): SQLAlchemy-compatible connection string to the PostgreSQL database.
         chunksize (int): Number of rows to process in each chunk.
     """
-    engine = create_engine(connection_string)
-    with open(parquet_file, "rb") as f:
-        table = pq.read_table(f)
-    df = table.to_pandas()
-
-    total_rows = len(df)
-    with tqdm(total=total_rows, desc="Loading to PostgreSQL", unit="rows") as pbar:
-        for start in range(0, total_rows, chunksize):
-            end = min(start + chunksize, total_rows)
-            chunk = df.iloc[start:end]
-            chunk.to_sql(
-                "space2stats", engine, if_exists="append", index=False, method="multi"
-            )
-            pbar.update(len(chunk))
+    table = pq.read_table(parquet_file)
+    with (
+        pg.connect(connection_string) as conn,
+        conn.cursor() as cur,
+        tqdm(total=table.num_rows, desc="Loading to PostgreSQL", unit="rows") as pbar,
+    ):
+        cur.adbc_ingest(TABLE_NAME, table.slice(0, 0), mode="replace")
```

Comment:
@alukach; replace mode creates the table if it does not exist. Slicing to 0,0 only writes the schema and then we can use efficient bulk ingesting of chunks. It also seems to have solved the failing tests.

Reply:
Nice! Clever with the slice.